-1

**Failed Tests**:   ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a74177/34709/summary.html
**COMMIT**: a7bfe29642c84f7beed86568446e265046bd6838
**CMSSW**: CMSSW_13_3_X_2023-09-12-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/42757/34709/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Local Products Rules ..... started
>> Local Products Rules ..... done
>> Creating project symlinks
>> Entering Package RecoParticleFlow/PFClusterProducer
>> Compile sequence completed for CMSSW CMSSW_13_3_X_2023-09-12-1100
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 1
+ eval scram build outputlog '&&' '(python3' /data/cmsbld/jenkins/workspace/ib-run-pr-tests/cms-bot/buildLogAnalyzer.py --logDir /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_3_X_2023-09-12-1100/tmp/el8_amd64_gcc11/cache/log/src '||' 'true)'
++ scram build outputlog
>> Entering Package RecoParticleFlow/PFClusterProducer
Entering library rule at src/RecoParticleFlow/PFClusterProducer/plugins
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_3_X_2023-09-12-1100/src/RecoParticleFlow/PFClusterProducer/plugins/Basic2DClusterForEachSeed.cc
</pre>

--- a/RecoParticleFlow/PFClusterProducer/interface/PFHCALDenseIdNavigator.h
+++ b/RecoParticleFlow/PFClusterProducer/interface/PFHCALDenseIdNavigator.h
@@ -230,7 +230,7 @@ public:
   // Obtain the neighbour's DetId
   const uint32_t getNeighbourDetId(const DetId detId, const uint32_t direction) {
     // desired order for PF: NORTH, SOUTH, EAST, WEST, NORTHEAST, SOUTHWEST, SOUTHEAST, NORTHWEST
-    if (detId == 0)
+    if (detId == DetId(0))
       return 0;
 
     if (direction == 0)                        // NORTH

